repository/user.go

```diff
@@ -26,11 +27,31 @@ func (r *userRepo) GetUsers(ctx context.Context) (users []domain.User, err error
 	return
 }
 
-func (r *userRepo) GetUser(ctx context.Context, id int) (user *domain.User, err error) {
+func (r *userRepo) GetUserById(ctx context.Context, id int) (user *domain.User, err error) {
+	result := r.DB.First(&user, "id=?", id)
+
+	if result.Error != nil {
+		err = fmt.Errorf("[repo.User.First] failed: %w", result.Error)
```

Comment:
```suggestion
		err = fmt.Errorf("[repo.User.GetUserById] failed: %w", result.Error)
```
Chỗ này là log lỗi trong hàm GetUserByID thì phải ghi ở trong error lỗi là GetUserById thì khi gặp lỗi mới trace được